### DISCUSSION TITLE
Upgrade Keycloak v#4 to latest

### DISCUSSION THREAD

**Discussion by islamzatary** [keycloak/keycloak#19072] [Q&A]
Hi,

i want to upgrade my keycloak version from v#4 to latest, already have users with multi tenancy and i want to upgrade to the latest to fix all vulnerability raised by pen test and very old documentation and community.

so can you please let me know best and safest way to start with this upgrade.

Thanks.

**Comment by sschu** [comment]
Please check the upgrading guide: https://www.keycloak.org/docs/latest/upgrading/
